### DISCUSSION TITLE
Behaviors in code behind

### DISCUSSION THREAD

**Discussion by DarkMortainius** [AvaloniaUI/Avalonia#14936] [Q&A]
Hi, I am trying to implement the work-around for UpdateSourceTrigger referenced here: #3754, but I need to do it in the code-behind, not in XAML. I couldn't find any documentation on how to do that, but I gave it a shot, but it appears I'm missing something. (If there is documentation somewhere, what is the link?) Essentially I'm trying to turn this XAML into code:

```XAML
<DataTemplate DataType="datapoint:DataPoint">
    <TextBox>
        <Interaction.Behaviors>
            <behaviors:LostFocusUpdateBindingBehavior Text="{Binding Text}"/>
        </Interaction.Behaviors>
    </TextBox>
</DataTemplate>
```
Here is my attempt, but it throws an error saying "Cannot find a DataContext to bind to." for the behavior.

```c#
var textBoxTemplate = new FuncDataTemplate<DataPoint>((_, _) => {
    var textBox = new TextBox();
    // Work-around for the fact that Avalonia doesn't support UpdateSourceTrigger yet.
    // TODO: Change to use UpdateSourceTrigger once Avalonia supports it
    var behaviors = new BehaviorCollection {
        new LostFocusUpdateBindingBehavior() {
            [!LostFocusUpdateBindingBehavior.TextProperty] = new Binding("Text")
        }
    };
    Interaction.SetBehaviors(textBox, behaviors);
    return textBox;
});
```
I realize this will be a moot point once 11.1 is released, but I'd like to get this working in the meantime. Thanks!

**Comment by timunie** [comment]
11.1.0-beta1 is out for testing.

**Reply by thevortexcloud** [reply]
Elaborating on this: 11.1 adds proper UpdateSourceTrigger support.

**Comment by thevortexcloud** [comment]
Have you tried using https://github.com/AvaloniaUI/Avalonia.Markup.Declarative? It will probably make things simpler if it helps at all.
